### PR TITLE
PHPC-2216: Generate _id field when inserting Document or PackedArray instances

### PR DESCRIPTION
**Pull request by alcaeus**
PHPC-2216

Test is named to not conflict with the new test added in #1415.

### CODE REVIEW

File: tests/bulk/bulkwrite-insert-003.phpt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+$manager = create_test_manager();
+
+$document = MongoDB\BSON\Document::fromPHP(['foo' => 'bar']);
+$array = MongoDB\BSON\PackedArray::fromPHP(['bar']);
+
+$bulk = new MongoDB\Driver\BulkWrite();
+
+$insertedDocumentId = $bulk->insert($document);
+$insertedArrayId = $bulk->insert($array);
```
Comment (jmikola) [inline tests/bulk/bulkwrite-insert-003.phpt:19]: Is there a use case for allowing `php_phongo_zval_to_bson_internal` to encode a PackedArray as a root document? I'm not sure if the code path is needed by anything in PHPC, or if you just added it because an array is technically just a `bson_t`.

Would it make sense to throw instead?

**Review by jmikola — COMMENTED** [review]
Code changes LGTM, but I do have a question about PackedArray.

File: tests/bulk/bulkwrite-insert-003.phpt (hunk 1 above)
Comment (alcaeus) [reply tests/bulk/bulkwrite-insert-003.phpt:19]: `php_phongo_zval_to_bson_internal` (and `php_phongo_zval_to_bson`) is used in several places, and we can't make an assumption that whatever we're converting will be a root document to be inserted to the database. If we want to prevent this, the correct course of action would be to catch this in the various `BulkWrite` operations, preventing the usage as a document in `insert`. `BulkWrite::update` will be a bit more complicated: while the replacement document shouldn't be a document directly, the `PackedArray` instance could also hold an aggregation pipeline to be used for the update, in which case we'd want to convert and send it to the server as is.

Long story short, I don't think this is something we should worry about. The main reason for the separation was to ignore the stored keys when iterating a `PackedArray` instance (as opposed to `Document` where they are relevant).

Comment (alcaeus) [reply tests/bulk/bulkwrite-insert-003.phpt:19]: Created PHPC-2219 to track this.
